### PR TITLE
Basic sessionState message handling

### PR DESCRIPTION
**Pull request by humanamburu**
```js
handleServerMessage({ 
	name: 'session.state',
        sessionId: 'rsconf-2017',
        participantId: 'abcd-1234-qwer-5678',
        puzzleIndex: 4,
        puzzleCount: 10,
        puzzle: {
            name: 'Puzzle name',
            input: '[{ ... }]',
            expected: '[1, 2, 3]'
        },
        roundPhase: 'in-progress',
        roundCountdown: 55, // s
        startCountdown: 0, // s
        score: {
            roundScore: [{
                participantId: 'abcd-1234-qwer-5678',
                time: 12, // s
                length: 37,
                correct: false
            }],
            aggregateScore: [{
                participantId: 'abcd-1234-qwer-5678',                    
                time: 12 // s
            }]
        }  
}, dispatch)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
   "dependencies": {
     "codemirror": "^5.23.0",
     "js-beautify": "^1.6.8",
-    "message-factory": "21-23/message-factory#v2.3.0",
+    "message-factory": "21-23/message-factory#v2.5.1",
```
Comment (kirilknysh) [inline package.json:40]: 1. yarn remove
2. yarn add

otherwise yarn.lock file is invalid

File: app/game-master/middleware/server-pipe.js [hunk 2]
```diff
@@ -10,12 +10,44 @@ import * as RoundActions from '../actions/round';
...
+        round: {
+            name,
+            input,
+            expected,
+            countdownRemaining: roundCountdown,
+            duration: startCountdown,
+            phase: roundPhase,
+        },
+        participant: {
+            participantId: message.participantId,
```
Comment (kirilknysh) [inline app/game-master/middleware/server-pipe.js]: https://github.com/21-23/message-factory/blob/master/app/protocol/ui.js#L48

 *-* participantId
 *+* displayName

File: package.json (hunk 1 above)
Comment (humanamburu) [reply package.json:40]: Thank you

File: app/game-master/middleware/server-pipe.js (hunk 2 above)
Comment (humanamburu) [reply app/game-master/middleware/server-pipe.js]: okay, i will update wiki

Comment (kirilknysh) [reply app/game-master/middleware/server-pipe.js]: wiki? it's correct there https://github.com/21-23/_qd/wiki/Joining-or-leaving-a-session

Comment (humanamburu) [reply app/game-master/middleware/server-pipe.js]: 🤕 yep, copy pasted message from redemption, sorry

**Comment by humanamburu** [comment]
Updated!

**Comment by kirilknysh** [comment]
LGTM, also I think `PARTICIPANT_INFO` could be removed as participant's info is a part of session state

**Review by kirilknysh — APPROVED** [review]

File: app/game-master/mock-data/current-round.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export default {
+    name: 'Current round',
+    duration: 120,
+    remaining: 95,
+    input: JSON.stringify([{ name: 'Johnie', surname: 'Walker', age: 14 }, { name: 'Johnie', surname: 'Walker', age: 20 },{ name: 'Adam', surname: 'Smith', age: 99 },{ name: 'Jack', surname: 'Daniels', age: 18 }]),
```
Comment (DaQuirm) [inline app/game-master/mock-data/current-round.js]: Johnnie is (usually) spelled with two `n`s.
Sorry 🐱

File: app/game-master/middleware/server-pipe.js [hunk 4]
```diff
@@ -10,12 +10,44 @@ import * as RoundActions from '../actions/round';
...
+        participant: {
+            displayName,
+        },
+        score: {
+            round: roundScore,
+            aggregate: aggregateScore,
+        },
+        session: {
+            currentRoundIndex: puzzleIndex,
+            puzzleCount: puzzleCount,
```
Comment (DaQuirm) [inline app/game-master/middleware/server-pipe.js]: Can be reduced to property shortcut (don't we have an eslint rule for that?), right?

Comment (humanamburu) [reply app/game-master/middleware/server-pipe.js]: Oh, sure

Comment (DaQuirm) [reply app/game-master/middleware/server-pipe.js]: Cool, what about eslint though?

**Review by DaQuirm — APPROVED** [review]

**Pull request merged**